### ISSUE TITLE
incompatible to some input method like Japanese or Chinese

### ISSUE THREAD

**Issue by weichenlin** [nextcloud/forms#2220]
**Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!**

**Describe the bug**
when add a checkbox or select options, if i type any char, a new empty input item will appear below.
this behavior will interrupt some input method like Chinese or Japanese, that one character needs multiple type,
and cause error message "There was an issue deleting this option", also the option value is incorrect.

**To Reproduce**

1. Click "new form"
2. Click "add a question", select "Dropdown"
3. Click "add a new answer" input field
4. use Japanese input method, type "sa" for char さ

**Expected behavior**
the char "さ" appear in the answer

**Screenshots**
![Screenshot_20240622_155814s](https://github.com/nextcloud/forms/assets/3406567/f3ab0de3-a62d-474c-8a29-6935031cb37e)

**Nextcloud (please complete the following information):**

-   Nextcloud-Version: [Nextcloud Hub 8 (29.0.2)]
-   Forms-Version: [4.2.4]

**Desktop (please complete the following information):**

-   OS: [Windows 11]
-   Browser [chrome]
-   Version [126.0.6478.115]

**Browser log**

```
NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2     DELETE https://nc2.mysite.net/ocs/v2.php/apps/forms/api/v2.4/option/cbrrt 400
(anonymous) @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
xhr @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
qi @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
_request @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
request @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
bs.<computed> @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
(anonymous) @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
deleteOptionFromDatabase @ Submit-DWdudiQx.mjs:2
deleteOption @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
checkValidOption @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
handleTabbing @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
focus @ Submit-DWdudiQx.mjs:2
focusIndex @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Ga.e.$nextTick @ vendor-DFuGXkIP.mjs:2
render @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
Ga.e._render @ vendor-DFuGXkIP.mjs:2
r @ vendor-DFuGXkIP.mjs:2
e.get @ vendor-DFuGXkIP.mjs:2
e.run @ vendor-DFuGXkIP.mjs:2
ws @ vendor-DFuGXkIP.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Rt @ vendor-DFuGXkIP.mjs:2
e.update @ vendor-DFuGXkIP.mjs:2
e.notify @ vendor-DFuGXkIP.mjs:2
set @ vendor-DFuGXkIP.mjs:2
m.<computed> @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
updateOptions @ Submit-DWdudiQx.mjs:2
addNewEntry @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2 [ERROR] forms: Error while deleting an option {app: 'forms', uid: 'admin', level: 2, option: {…}, error: V}
value @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
value @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
Promise.catch (async)
deleteOptionFromDatabase @ Submit-DWdudiQx.mjs:2
deleteOption @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
checkValidOption @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
handleTabbing @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
focus @ Submit-DWdudiQx.mjs:2
focusIndex @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Ga.e.$nextTick @ vendor-DFuGXkIP.mjs:2
render @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
Ga.e._render @ vendor-DFuGXkIP.mjs:2
r @ vendor-DFuGXkIP.mjs:2
e.get @ vendor-DFuGXkIP.mjs:2
e.run @ vendor-DFuGXkIP.mjs:2
ws @ vendor-DFuGXkIP.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Rt @ vendor-DFuGXkIP.mjs:2
e.update @ vendor-DFuGXkIP.mjs:2
e.notify @ vendor-DFuGXkIP.mjs:2
set @ vendor-DFuGXkIP.mjs:2
m.<computed> @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
updateOptions @ Submit-DWdudiQx.mjs:2
addNewEntry @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
Submit-DWdudiQx.mjs:2 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'value')
    at s.onInput (Submit-DWdudiQx.mjs:2:293202)
onInput @ Submit-DWdudiQx.mjs:2
await in onInput (async)
(anonymous) @ Submit-DWdudiQx.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Ga.e.$nextTick @ vendor-DFuGXkIP.mjs:2
render @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
Ga.e._render @ vendor-DFuGXkIP.mjs:2
r @ vendor-DFuGXkIP.mjs:2
e.get @ vendor-DFuGXkIP.mjs:2
e.run @ vendor-DFuGXkIP.mjs:2
ws @ vendor-DFuGXkIP.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Rt @ vendor-DFuGXkIP.mjs:2
e.update @ vendor-DFuGXkIP.mjs:2
e.notify @ vendor-DFuGXkIP.mjs:2
set @ vendor-DFuGXkIP.mjs:2
m.<computed> @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
updateOptions @ Submit-DWdudiQx.mjs:2
addNewEntry @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
```

**Comment by TUCAOEVER** [comment]
https://github.com/nextcloud/forms/assets/43756134/ea26374b-6d27-44fe-a5bf-e4cc68003ebf

To imporve understanding, i made a vedio about this issue.

**Comment by Chartman123** [comment]
Ok, thank you. I've already got a solution for it in mind :) Let's see if it really works...

**Comment by Chartman123** [comment]
@weichenlin @TUCAOEVER do you have a development environment where you could check my PR?

**Comment by TUCAOEVER** [comment]
> @weichenlin @TUCAOEVER do you have a development environment where you could check my PR?

Sorry, I don't have it. Could you possibly pack it as an app so I can try to install it manually and test how it works?

**Comment by Chartman123** [comment]
Yes, I can, but please be aware that there are some other changes included, that are not yet released and that might change your database structure for Forms. So it's not possible to go back to the currently released version without manually adjusting your database again.

**Comment by TUCAOEVER** [comment]
> Yes, I can, but please be aware that there are some other changes included, that are not yet released and that might change your database structure for Forms. So it's not possible to go back to the currently released version without manually adjusting your database again.

Don't worry about that, i have set up a individual docker for testing, so it doesn't matter if it changed the db structure.

**Comment by Chartman123** [comment]
Ok, fine... Just wanted to make sure. 😉 

For all others: Please only download and install it if you really know what you do!

[forms.tar.gz](https://github.com/user-attachments/files/16105664/forms.tar.gz)

@TUCAOEVER please test in different browsers. So far I found differences in handling the input events between Firefox and Chromium based browsers.

**Comment by Chartman123** [comment]
@TUCAOEVER I uploaded a new version to the link above... You could try again, I think I tackled the remaining inconsistencies

**Comment by TUCAOEVER** [comment]
> @TUCAOEVER I uploaded a new version to the link above... You could try again, I think I tackled the remaining inconsistencies

Just have tested it, the dropdown or checkbox works totally fine, nice work! But meanwhile i found someting weired about the new feature about "Upload file" option, and you can see in screenshots below
![image](https://github.com/nextcloud/forms/assets/43756134/d60cab54-cb02-463b-ad67-915054c2830a)
In editor view, there is spacing between the option and answer.
![image](https://github.com/nextcloud/forms/assets/43756134/02ac0a90-8954-48cd-a6f9-757777fde579)
In preview page, there is spacing under the upload file button, which may related to the layout issue.
Though it doen't effect using but really looks not that perfect, hope you could double check the issue.

**Comment by TUCAOEVER** [comment]
> > @TUCAOEVER I uploaded a new version to the link above... You could try again, I think I tackled the remaining inconsistencies
> 
> Sry for late reply, busy doing my job last week, i just have time to test it, the dropdown or checkbox works fine but have a little detail which i don't know if it a bug or your designed,
https://github.com/nextcloud/forms/assets/43756134/526dc268-f436-4456-b441-6dfcbf721382
when u start typing the first answer, before entering the new option will be automaticlly created, but the second option doesn't look like the first one, as u can see in the vedio, only after i have typed something in the box, the new option can be created, expect for that, nice work!
Meanwhile i found someting weired about the new feature about "Upload file" option, and you can see in screenshots below ![image](https://private-user-images.githubusercontent.com/43756134/347216691-d60cab54-cb02-463b-ad67-915054c2830a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1NzQ1OTcsIm5iZiI6MTcyMDU3NDI5NywicGF0aCI6Ii80Mzc1NjEzNC8zNDcyMTY2OTEtZDYwY2FiNTQtY2IwMi00NjNiLWFkNjctOTE1MDU0YzI4MzBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDAxMTgxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwNmI1YjI3YTA0NzU1NGYxOTBkODA1MTMzYWMzNzJhM2U3ZTU3YTI5NWFmYTliNTMzYzgwZDUyZmIwMThjMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZRxBUNmJ5Nd8TVsFc3sciV4BwamXlnDhgPR5gn46TfM) In editor view, there is spacing between the option and answer. ![image](https://private-user-images.githubusercontent.com/43756134/347217076-02ac0a90-8954-48cd-a6f9-757777fde579.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1NzQ1OTcsIm5iZiI6MTcyMDU3NDI5NywicGF0aCI6Ii80Mzc1NjEzNC8zNDcyMTcwNzYtMDJhYzBhOTAtODk1NC00OGNkLWE2ZjktNzU3Nzc3ZmRlNTc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDAxMTgxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjYTgxNjY4ZDY4OWE4NGFjMTJmMGQwNGZkNDdlOGI5MWJiZWNlOTgwMTIyOTNhMjRkY2MxMmIxOTc1OTgzNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1apS-P-vzUdOUCXIwW_65GKkRk7T6Hvh0A0xC22Sem0) In preview page, there is spacing under the upload file button, which may related to the layout issue. Though it doen't effect using but really looks not that perfect, hope you could double check it.

**Comment by Chartman123** [comment]
Could you please create a new issue for that file question stuff?

**Comment by Chartman123** [comment]
@TUCAOEVER ok, I did another check... there's a slight difference in how the first option is handled... when you create a new question, the "Add a new answer" field is shown, which is more just like a dummy field. as soon as you enter some text, it will move the text to a new field (answer number 1), and so on.

When you delete all existing answers/options, the last field (answer number 1) will stay there. This is then just like every other answer field. it will show the "Add a new answer" dummy field as soon as some text is added to field 1.

So basically everything "as expected". 👍🏻

**Comment by TUCAOEVER** [comment]
> Could you please create a new issue for that file question stuff?
> 
> Glad to hear that the IME things are working now :) And I'll have to investigate the use case with the first option of a question... I currently only tested it with questions where other options already existed.

Of course!
